platform/pulp/tasking/services/storage.py

```diff
@@ -137,3 +137,13 @@ def _rmtree_fix_permissions(directory_path):
             shutil.rmtree(directory_path)
         else:
             raise
+
+@contextmanager
+def working_dir_context():
+    """
+    Creates a context manager for a working that needs a working directory.
+    """
+    working_dir = get_working_directory()
+    os.chdir(working_dir)
+    yield working_dir
+    delete_working_directory()
```

Comment:
If L#148 raises an exception I think we still want the working dir to be deleted regardless. What do you think having the `yield` in a `try` block and having the `delete_working_directory()` in a `finally` block?